app/src/main/java/org/mozilla/fenix/settings/CookieBannerFragment.kt

```diff
@@ -0,0 +1,131 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings
+
+import android.os.Bundle
+import androidx.preference.CheckBoxPreference
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import mozilla.components.concept.engine.EngineSession.CookieBannerHandlingMode
+import mozilla.components.concept.engine.Settings
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.ext.showToolbar
+import org.mozilla.fenix.utils.view.addToRadioGroup
+
+/**
+ * Lets the user set up the cookie banner handling preferences.
+ */
+class CookieBannerFragment : PreferenceFragmentCompat() {
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.cookie_banner_preferences, rootKey)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        showToolbar(getString(R.string.preferences_cookie_banner_reduction))
+        setupPreferences()
+    }
+
+    private fun getEngineSettings(): Settings {
+        return requireContext().components.core.engine.settings
+    }
+
+    private fun setupPreferences() {
+        requirePreference<CheckBoxPreference>(R.string.pref_key_cookie_banner_private_browsing).apply {
+            isChecked = context.settings().cookieBannerHandlingPrivateBrowsing
```

Comment:
var name `isChecked` could be more specific like `enabledInPrivateBrowsing`. also I still think this var is unnecessary and can be removed